### PR TITLE
`Development`: Improve client test coverage for exam management components

### PR DESCRIPTION
**Pull request by coaxial1**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context

Increase coverage for ``app/exam/manage``.

### Description

Added and extended several tests. Some parts that are not be tested or do not contain logic have been excluded from the coverage report, i.e. route definitions or methods that just fire a HTTP call, but do not contain any logic.

Coverage of whole ``app/exam/manage`` (Line, Branch):
-- Before: ``51.42%, 42.14%``
-- After: ``97.48%, 72.38%``

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

**Review by willergeorg — DISMISSED** [review]
You have one unused import in `student-exam.service.spec.ts`. Besides that, test code lgtm.

File: src/main/webapp/app/exam/manage/exercise-groups/exercise-group.service.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export class ExerciseGroupService {
 
     constructor(private router: Router, private http: HttpClient) {}
 
+    /* istanbul ignore next */
```
Comment (b-fein) [inline src/main/webapp/app/exam/manage/exercise-groups/exercise-group.service.ts:16]: I dont’t think it’s worth artificially increasing the coverage by excluding those seven (four here, three in the other file) statements. I think it’s acceptable if they are just not tested in this case.

**Review by b-fein — DISMISSED** [review]
The code changes look good, but I added two inline comments where I think small improvements can be made.

File: src/main/webapp/app/exam/manage/exercise-groups/exercise-group.service.ts (hunk 1 above)
Comment (coaxial1) [reply src/main/webapp/app/exam/manage/exercise-groups/exercise-group.service.ts:16]: 🤷 Without them, they create a big fat red line in the coverage report ...
I've removed them anyways

**Review by akesfeden — COMMENTED** [review]
> I think it would be better to use the assertions specific to spies. That would also be consistent with most of the other test code and the `.not.toHaveBeenCalled()` further down.

In order to be consistent and because of the reasons explained by @b-fein, I think it would be nice to convert other similar usages of `... .mock.calls).toHaveLength` and `... .mock.calls[0]).toEqual` to `.toHaveBeenCalledTimes` and `.toHaveBeenCalledWith` as well. Searching `.mock.calls).toHaveLength` in the [Files Changed](https://github.com/ls1intum/Artemis/pull/4607/files) section shows 5 usages. Other than that, tests look good to me.

**Comment by coaxial1** [comment]
I've removed the occurrences of these things where possible. Sometimes I need to access the parameters of the calls directly to assert a few things on them.

**Review by b-fein — APPROVED** [review]
Thanks for the changes, looks good now. :+1: 

Using the remaining direct parameter accesses are fine, I think, as using further `expect.any()` or similar inside the `toHaveBeenCallediWith(…)` would probably not make the assertions more readable.

**Review by akesfeden — APPROVED** [review]
LGTM 👍

**Pull request merged**